planning/src/parsing/sexpr.rs

```diff
@@ -339,7 +351,27 @@ fn read(tokens: &mut std::iter::Peekable<core::slice::Iter<Token>>, src: &std::s
                 }
             }
         }
+
         Some(Token::RParen(_)) => bail!("Unexpected closing parenthesis"),
+        Some(quotting) => {
+            let (sym_quote, start) = match quotting {
+                Token::Quote(s) => (Sym::new("quote"), s),
+                Token::QuasiQuote(s) => (Sym::new("quasiquote"), s),
+                Token::Unquote(s) => (Sym::new("unquote"), s),
+                _ => bail!("Unexpected token, should be Quote, QuasiQuote or Unquote"),
+            };
+
+            let mut es = vec![SExpr::Atom(sym_quote)];
+            let e = read(tokens, src)?;
+            //let loc = e.loc();
+            es.push(e);
+            //Compute the span
+            Ok(SExpr::List(SList {
+                list: es,
+                source: src.clone(),
+                span: Span::new(*start, *start), //TODO: find a way to declare the span
```

Comment:
need to fix this and the the tests for contextual display.
My intuition is that the span should finish at the end of `e`

Reply:
The function to get the end of `e` returns a Loc object.
```
e.loc().end() -> Loc.
```
From this **Loc** we cannot get the **end** Pos

Reply:
Essentially, you need `e.span.end`. 
`span` is private is this case but you can add an accessor for the start and end `Pos` of a `Loc`